models/solution/stop.rb

```diff
@@ -60,6 +60,10 @@ def initialize(object, options = {})
       end
 
       def vrp_result(options = {})
+        if info.travel_time&.positive?
```

Comment:
If I understood it correctly, you moved the setup_duration "parsing"/"correction" to the very end so that we don't have to worry about correcting (applying, rewinding, etc) this info. Is that it?

Reply:
Exact !